### PR TITLE
Rnog data example

### PR DESCRIPTION
**Pull request by iplaisier**
Example to read RNO_G data

### CODE REVIEW

File: NuRadioReco/examples/RNO_data/read_data_example/run_reconstruction.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+""" An example to show how to read RNO-G data and how to perform simple reconstructions. The data used is pulser data and a simple (brute force, not optimized) spherical wave reconstruction is performed to obtain the pulser position """""
+
+""" Initiazize modules needed"""
+
+channelBandPassFilter = NuRadioReco.modules.channelBandPassFilter.channelBandPassFilter()
+sphericalWaveFitter = NuRadioReco.modules.sphericalWaveFitter.sphericalWaveFitter()
+channelAddCableDelay = NuRadioReco.modules.channelAddCableDelay.channelAddCableDelay()
+
+
+use_channels = [10,9, 0,3, 21, 22]
```
Comment (shallmann) [inline NuRadioReco/examples/RNO_data/read_data_example/run_reconstruction.py:22]: I don't know which one you want to keep, but except for the `use_channels` line `NuRadioReco/examples/RNO_data/read_data_example/run_reconstruction.py` and `NuRadioReco/examples/RNO_data/run_reconstruction.py` are identical

(if you keep this one, maybe you want to add some spaces for beauty `use_channels = [10, 9, 0, 3, 21, 22]`

File: NuRadioReco/examples/RNO_data/run_reconstruction.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        i = 1
+        for channel in station.iter_channels():
+            if channel.get_id() in use_channels:
+                ax = fig.add_subplot(3, 2, i)
+                ax.plot(channel.get_trace())
+                ax.title.set_text("channel id {}".format(channel.get_id()))
+                ax.grid()
+                i+= 1
+
+    fig.tight_layout()
```
Comment (shallmann) [inline NuRadioReco/examples/RNO_data/run_reconstruction.py:65]: indent lines 65 and 66 (should be inside the `if plots`)

File: NuRadioReco/examples/RNO_data/run_reconstruction.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+det = detector.Detector(json_filename = "detector_description.json")
+det.update(datetime.datetime(2022, 10, 1))
+
+""" Get positions for the pulsers from the detector file as starting positions for the fit """
+
+station_id = 21
+pulser_id = 3  #Helper string C
+pulser_position = det.get_relative_position(station_id, pulser_id, mode = 'device')
+rel_pulser_position = [pulser_position[0], pulser_position[1], pulser_position[2]]
```
Comment (shallmann) [inline NuRadioReco/examples/RNO_data/run_reconstruction.py:35]: isn't the return of get_relative_position already a numpy array? Probably one doesn't need this line

File: NuRadioReco/modules/sphericalWaveFitter.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            return -1*corr
+            
+            
+
+        trace = np.copy(station.get_channel(self.__channel_pairs[0][0]).get_trace())
+        self.__correlation = np.zeros((len(self.__channel_pairs), len(np.abs(scipy.signal.correlate(trace, trace))) ))
+
+        for ich, ch_pair in enumerate(self.__channel_pairs):
+            trace1 = np.copy(station.get_channel(self.__channel_pairs[ich][0]).get_trace())
+            trace2 =np.copy(station.get_channel(self.__channel_pairs[ich][1]).get_trace())
```
Comment (shallmann) [inline NuRadioReco/modules/sphericalWaveFitter.py:92]: `=np.copy` -> `= np.copy` for beauty

File: NuRadioReco/modules/sphericalWaveFitter.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+            tmp = get_time_delay_spherical_wave(start_pulser_position, ch_pair, n=n_index)
+            n_samples = tmp * self.__sampling_rate
+            self._pos_starting[ich] = int(len(self.__correlation[ich]) / 2 - n_samples)
+
+
+        method = 'Nelder-Mead'
+        x_start, y_start, z_start = start_pulser_position
+        dx, dy, dz = [.1, .1, .1]
+        #ll = opt.minimize(likelihood, x0 = (start_pulser_position[0]-10, start_pulser_position[1], start_pulser_position[2]),method = method)#
```
Comment (shallmann) [inline NuRadioReco/modules/sphericalWaveFitter.py:121]: do you need this? else maybe a comment why you tried it out and why you opted for the other one.

File: NuRadioReco/modules/sphericalWaveFitter.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        for ich, ch_pair in enumerate(self.__channel_pairs):
+            positions = self.__relative_positions[ich]
+            times = []
+
+            tmp = get_time_delay_spherical_wave(start_pulser_position, ch_pair, n=n_index)
+            n_samples = tmp * self.__sampling_rate
+            self._pos_starting[ich] = int(len(self.__correlation[ich]) / 2 - n_samples)
+
+
+        method = 'Nelder-Mead'
```
Comment (shallmann) [inline NuRadioReco/modules/sphericalWaveFitter.py:118]: the line where this is used is commented. Needed? In the `if debug` it is redefined.

**Review by shallmann — CHANGES_REQUESTED** [review]
Great work, thanks for implementing this. Please remove the duplicated example file and indent the lines in run_reconstruction. My other comments are cosmetic

File: NuRadioReco/examples/RNO_data/run_reconstruction.py (hunk 3 above)
Comment (iplaisier) [reply NuRadioReco/examples/RNO_data/run_reconstruction.py:35]: True

**Review by shallmann — APPROVED** [review]
Thanks for making these changes!

**Pull request merged**